### PR TITLE
Refactor BuildFragmentCatalog - 2

### PR DESCRIPTION
**Pull request by gedeck**
Reviewed the unit test code:

- replaced assert with unittest assert methods
- Enable manually excluded longer test to be included using a command line flag

Extracted the progress reporting into a separate class. The intention was to reduce the amount of boilerplate code dedicated to this to help focusing on the actual tasks.

### CODE REVIEW

File: rdkit/Chem/UnitTestCatalog.py [hunk 1]
```diff
@@ -158,86 +159,98 @@ def _test5MoreComplex(self):
...
+  import argparse
+  import sys
+  parser = argparse.ArgumentParser()
+  parser.add_argument('-l', default=False, action='store_true', dest='doLong')
+  args = parser.parse_args()
+  doLong = args.doLong
 
-if __name__ == '__main__':
+  # Remove the -l flag if present so that it isn't interpreted by unittest.main()
+  if 'l' in sys.argv:
```
Comment (greglandrum) [inline rdkit/Chem/UnitTestCatalog.py]: shouldn't that be `-l`?

Comment (gedeck) [inline rdkit/Chem/UnitTestCatalog.py]: Yes and I have fixed a few more files where I introduced the same mistake.

File: rdkit/Chem/Draw/UnitTestDraw.py [hunk 2]
```diff
@@ -152,6 +152,28 @@ def testSpecialCases(self):
...
+    heteroatom = Chem.MolFromSmarts('[#7,#8,#15,#16]')  # .GetAtomWithIdx(0)
+    halogen = Chem.MolFromSmarts('[#9,#17,#35,#53]')  # .GetAtomWithIdx(0)
+    mol = Chem.MolFromSmarts('[#7,#8,#15,#16]=C(CC1=CC([#9,#17,#35,#53])=CC=C1)[#7,#8,#15,#16]')
+    for match in mol.GetSubstructMatches(heteroatom):
+      print(match)
+      mol.GetAtomWithIdx(match[0]).SetProp('drawlabel', 'Q')
+    for match in mol.GetSubstructMatches(halogen):
+      print(match)
+      mol.GetAtomWithIdx(match[0]).SetProp('drawlabel', 'X')
+    self._testMolToImage(mol=mol, showImage=True, kekulize=True)
```
Comment (greglandrum) [inline rdkit/Chem/Draw/UnitTestDraw.py]: Probably you don't want 'showImage=True' here

File: rdkit/Chem/Draw/UnitTestDraw.py [hunk 3]
```diff
@@ -152,6 +152,28 @@ def testSpecialCases(self):
...
+    
+    img = Draw.MolToImage(mol, size=(300, 300))
+    img.show()
+
```
Comment (greglandrum) [inline rdkit/Chem/Draw/UnitTestDraw.py]: Or this call to show()

**Comment by gedeck** [comment]
I close this one and open a new one. The change to UnitTestDraw shouldn't have been in this PR.

**Pull request closed without merging**